### PR TITLE
test: migrate packages to jest

### PR DESCRIPTION
**Pull request by juanpicado**
On going jest migration https://github.com/pnpm/pnpm/issues/2858

- [x]  /pnpm/packages/pnpmfile
- [x]  /pnpm/packages/package-bins
- [x]  /pnpm/packages/outdated

### CODE REVIEW

File: packages/pnpmfile/test/index.ts [hunk 1]
```diff
@@ -1,35 +1,24 @@
-import { requirePnpmfile } from '@pnpm/pnpmfile'
+import { requirePnpmfile, BadReadPackageHookError } from '@pnpm/pnpmfile'
```
Comment (juanpicado) [inline packages/pnpmfile/test/index.ts:1]: I need expose `BadReadPackageHookError` to have a more reliable expectation, `toThrow()` only is able to catch either an object or a message and I wanna verify the error code.

**Comment by juanpicado** [comment]
I'll take care of the issue, need to check why is failing.

**Comment by juanpicado** [comment]
@zkochan any idea why is failing?

**Comment by zkochan** [comment]
I don't know

**Comment by statianzo** [comment]
> @zkochan any idea why is failing?

It's as if there's a stale package.json, the CI is still running `_test:tap` for `outdated`

Screen Shot 2020-10-25 at 22 19 34

https://travis-ci.com/github/pnpm/pnpm/jobs/407008430#L2576

**Comment by zkochan** [comment]
oh, I see. There is this script: https://github.com/pnpm/pnpm/tree/master/utils/updater

it automatically updates the package.json files. So it changes the script back to using tape. I'll have to change it so that it fails if changes are tried to be done on CI. Also, it should not break packages that are migrated to Jest. I'll push a fix.

File: packages/pnpmfile/test/index.ts (hunk 1 above)
Comment (zkochan) [reply packages/pnpmfile/test/index.ts:1]: Great, this is much better

File: packages/outdated/test/index.ts [hunk 2]
```diff
@@ -1,3 +1,2 @@
-/// <reference path="../../../typings/index.d.ts"/>
 import './getManifest.spec'
 import './outdated.spec'
```
Comment (zkochan) [inline packages/outdated/test/index.ts]: this file should probably be removed

**Comment by juanpicado** [comment]
> oh, I see. There is this script: https://github.com/pnpm/pnpm/tree/master/utils/updater
> 
> it automatically updates the package.json files. So it changes the script back to using tape. I'll have to change it so that it fails if changes are tried to be done on CI. Also, it should not break packages that are migrated to Jest. I'll push a fix.

thanks for looking in to that, that's the thing I could not find. 🙇

**Pull request merged**